### PR TITLE
Make support for +p, ++p, +n, ++n, ++np consistent for all Charmruns

### PR DESCRIPTION
**Pull request by evan-charmworks**

### CODE REVIEW

**Review by stwhite91 — DISMISSED** [review]

File: src/util/charmrun-src/charmrun.C [hunk 1]
```diff
@@ -1198,14 +1198,14 @@ static void arg_init(int argc, const char **argv)
...
-        fprintf(stderr, "Charmrun> Error: ++ppn (number of PEs per node) does not divide +p (number of PEs).\n");
+        fprintf(stderr, "Charmrun> Error: ++ppn %d (number of PEs per node) does not divide +p %d (number of PEs).\n", arg_ppn, arg_requested_pes);
         exit(1);
       }
     }
 #else
     if (arg_requested_pes > 0 && arg_requested_nodes > 0 && arg_requested_pes != arg_requested_nodes)
     {
-      fprintf(stderr, "Charmrun> Error: +p and ++np do not agree.\n");
+      fprintf(stderr, "Charmrun> Error: +p %d and ++np %d do not agree.\n", arg_requested_pes, arg_requested_nodes);
```
Comment (jcphill) [inline src/util/charmrun-src/charmrun.C:1208]: Should be ++n/++np

File: src/util/charmrun-src/charmrun.C [hunk 2]
```diff
@@ -1185,7 +1185,7 @@ static void arg_init(int argc, const char **argv)
 #if CMK_SMP
     if (arg_requested_pes > 0 && arg_requested_nodes > 0 && arg_ppn > 0 && arg_ppn * arg_requested_nodes != arg_requested_pes)
     {
-      fprintf(stderr, "Charmrun> Error: ++np times ++ppn does not equal +p.\n");
+      fprintf(stderr, "Charmrun> Error: +n/++np %d times ++ppn %d does not equal +p %d.\n", arg_requested_nodes, arg_ppn, arg_requested_pes);
```
Comment (jcphill) [inline src/util/charmrun-src/charmrun.C:1188]: Should be ++n instead of +n.

**Review by jcphill — COMMENTED** [review]
I only care about charmrun.C as I've never used the bash charmrun scripts.

File: src/util/charmrun-src/charmrun.C (hunk 1 above)
Comment (evan-charmworks) [reply src/util/charmrun-src/charmrun.C]: Fixed

File: src/util/charmrun-src/charmrun.C (hunk 2 above)
Comment (evan-charmworks) [reply src/util/charmrun-src/charmrun.C:1188]: +n with a single plus is supported, like +p. I don't want to change all mentions of +p and +n to also say ++p and ++n.

Comment (jcphill) [reply src/util/charmrun-src/charmrun.C:1188]: OK, thanks.

**Review by matthiasdiener — APPROVED** [review]

**Review by stwhite91 — APPROVED** [review]

**Pull request merged**